src/components/implementation/no_interface/vkernel/micro_booter.c

```diff
@@ -17,25 +17,31 @@ int vmid;
 void
 vm_init(void *d)
 {
-	vmid = (int)d;
+	int         rcvd, blocked;
+	cycles_t    cycles;
+	thdid_t     tid;
+	tcap_time_t timeout = 0, thd_timeout;
+
+	vmid = cos_sinv(VM_CAPTBL_SELF_SINV_BASE, VK_SERV_VM_ID << 16 | cos_thdid(), 0, 0, 0);
```

Comment:
I wanted to set a protocol for invocation, where the second parameter is "option" + "initthd id of vm requesting it", nothing to do with the unused arguments, perhaps if I extend this SINV, will need those for something or maybe not.